File: JavaScript/MaximumDistanceInArrays/MaximumDistanceInArrays.js
/**
 * (128ms)
 * @param {number[][]} arrays
 * @return {number}
 */
var maxDistance = function(arrays) {
    var res = 0;
    var min = arrays[0][0], max = arrays[0][arrays[0].length - 1];
    for (var i = 1; i < arrays.length; i++) {
    	res = Math.max(res, max - arrays[i][0], arrays[i][arrays[i].length - 1] - min);
    	min = Math.min(min, arrays[i][0]);
    	max = Math.max(max, arrays[i][arrays[i].length - 1]);
    }
    return res;
};

File: Python/MaximumDistanceInArrays/MaximumDistanceInArrays.py
class Solution(object):
	# (82ms)
    def maxDistance(self, arrays):
        """
        :type arrays: List[List[int]]
        :rtype: int
        """
        res, min_now, max_now = 0, arrays[0][0], arrays[0][-1]
        for array in arrays[1:]:
        	res = max(res, max_now - array[0], array[-1] - min_now)
        	min_now, max_now = min(min_now, array[0]), max(max_now, array[-1])
        return res